trlc/ast.py

```diff
@@ -3082,6 +3084,24 @@ def all_names(self):
             rv |= self.parent.all_names()
         return rv
 
+    def iter_record_objects_by_section(self):  
+        section_dictionary = {}        
+        for record_object in self.iter_record_objects():           
+            location = record_object.location.file_name        
+            if location not in self.trlc_files:
+                self.trlc_files.append(location)
+                yield location            
+            if record_object.section:
+                self.section_names = record_object.section
+                if len(self.section_names) > 0:
+                    for level in range(len(self.section_names)):
+                        if self.section_names[level].name not in section_dictionary.keys():
+                            section_dictionary.update({self.section_names[level].name: True})                            
+                            yield self.section_names[level].name, level
+                    yield record_object.name
+            else:
+                yield record_object.name
```

Comment:
a proposal for refactoring the code:
get rid of self.section_dictionary then. not necessary
```suggestion
        object_level = len(record_object.section)-1 if record_object.section else 0
        if record_object.section:
            for level, section in enumerate(record_object.section):
                if section.name not in section_names:
                    section_names.append(section.name)
                    yield section.name, level
            yield record_object, object_level
        else:
            yield record_object, object_level
```